### PR TITLE
Hotfix/adding policy statement for secretsmanager

### PR DESCRIPTION
**Pull request by Amar-NHS**
## Description

Added permissions to retrieve the secrets and refactored the lambda to receive the CIS2 client ID through secrets manager.

## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Review by chenglawson1 — APPROVED** [review]

File: terraform/environment/dev/dev-4.tfvars [hunk 1]
```diff
@@ -4,5 +4,5 @@ dns_zone                = "cicd"
 CIS2_KNAME              = "CIS2_INT_1"
 CIS2_CLIENT_ID          = ""
```
Comment (chenglawson1) [inline terraform/environment/dev/dev-4.tfvars:5]: See above.

File: terraform/environment/dev/dev-5.tfvars [hunk 2]
```diff
@@ -3,5 +3,5 @@ dns_zone                = "cicd"
 CIS2_KNAME              = "CIS2_INT_1"
 CIS2_CLIENT_ID          = ""
```
Comment (chenglawson1) [inline terraform/environment/dev/dev-5.tfvars:4]: See above.

**Review by Abdul-NHS — APPROVED** [review]

**Review by Abdul-NHS — DISMISSED** [review]

**Review by avi312NHS — APPROVED** [review]

**Review by MichaelW-NHS — DISMISSED** [review]

File: terraform/environment/dev/dev-1.tfvars [hunk 3]
```diff
@@ -3,5 +3,5 @@ dns_zone                = "cicd"
 CIS2_KNAME              = "CIS2_INT_1"
 CIS2_CLIENT_ID          = ""
```
[the remark this answers is not in the document]
Comment (Amar-NHS) [reply terraform/environment/dev/dev-1.tfvars]: its aligned on other environment files

File: terraform/environment/dev/dev-5.tfvars (hunk 2 above)
Comment (Amar-NHS) [reply terraform/environment/dev/dev-5.tfvars]: resolved

**Review by chenglawson1 — APPROVED** [review]

**Review by Abdul-NHS — APPROVED** [review]

File: terraform/environment/nft/nft.tfvars [hunk 4]
```diff
@@ -2,7 +2,7 @@
 environment             = "nft"
 dns_zone                = "nft"
 CIS2_KNAME              = "CIS2_INT_1"
-CIS2_CLIENT_ID          = ""
+CIS2_ID                 = "CIS2_CLIENT_ID"
 CIS2_TOKEN_ENDPOINT_URL = "https://am.nhsint.auth-ptl.cis2.spineservices.nhs.uk:443/openam/oauth2/realms/root/realms/NHSIdentity/realms/Healthcare/access_token"
 CIS2_PUBLIC_KEY_ID      = "CIS2_INT_1"
```
[the remark this answers is not in the document]
Comment (Amar-NHS) [reply terraform/environment/nft/nft.tfvars]: resolved

**Pull request merged**